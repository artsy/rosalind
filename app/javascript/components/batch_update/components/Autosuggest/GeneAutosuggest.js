import React from 'react'
import PropTypes from 'prop-types'
import GenericAutosuggest from './GenericAutosuggest'
import { matchGenes } from 'lib/rosalind-api'
import { getSuggestionValue, renderSuggestion } from './helpers'

function GeneAutosuggest(props) {
  return (
    <GenericAutosuggest
      id="gene-autosuggest"
      placeholder={props.placeholder}
      fetchSuggestions={matchGenes}
      getSuggestionValue={getSuggestionValue}
      renderSuggestion={renderSuggestion}
      selectSuggestion={gene => {
        props.onSelectGene(gene)
      }}
    />
  )
}

GeneAutosuggest.propTypes = {
  onSelectGene: PropTypes.func.isRequired,
}

GeneAutosuggest.defaultProps = {
  placeholder: 'Select a gene',
}

export { GeneAutosuggest }
