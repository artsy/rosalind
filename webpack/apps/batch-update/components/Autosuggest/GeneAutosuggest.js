import React from 'react'
import GenericAutosuggest from './GenericAutosuggest'
import { matchGenes } from 'lib/rosalind-api'
import { getSuggestionValue, renderSuggestion } from './helpers'

function GeneAutosuggest (props) {
  return (
    <GenericAutosuggest
      id='gene-autosuggest'
      placeholder={props.placeholder}
      fetchSuggestions={matchGenes}
      getSuggestionValue={getSuggestionValue}
      renderSuggestion={renderSuggestion}
      selectSuggestion={gene => { props.onSelectGene(gene) }}
      />
  )
}

GeneAutosuggest.propTypes = {
  onSelectGene: React.PropTypes.func.isRequired
}

GeneAutosuggest.defaultProps = {
  placeholder: 'Select a gene'
}

export { GeneAutosuggest }
