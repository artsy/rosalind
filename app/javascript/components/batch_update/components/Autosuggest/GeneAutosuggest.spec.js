import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { GeneAutosuggest } from './GeneAutosuggest'

const mockHandler = jest.fn()

it('renders correctly', () => {
  const rendered = renderer.create(<GeneAutosuggest onSelectGene={mockHandler} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
