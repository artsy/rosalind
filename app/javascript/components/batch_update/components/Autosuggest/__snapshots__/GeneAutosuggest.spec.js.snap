// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="GenericAutosuggest"
>
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-owns="react-autowhatever-gene-autosuggest"
    className="react-autosuggest__container"
    role="combobox"
  >
    <input
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-controls="react-autowhatever-gene-autosuggest"
      autoComplete="off"
      className="react-autosuggest__input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select a gene"
      type="text"
      value=""
    />
    <div
      className="react-autosuggest__suggestions-container"
      id="react-autowhatever-gene-autosuggest"
      role="listbox"
    />
  </div>
</div>
`;
